Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/AdaptiveHoughTransformSeeder.hpp

```diff
@@ -0,0 +1,478 @@
+  /// @param function is callable used to check crossing at the edges
+  template <typename F>
+  inline bool isLineInside(F function) const {
+    const float yB = function(m_xBegin);
+    const float yE = function(m_xBegin + m_xSize);
+    return (yE > yB) ? yB < m_yBegin + m_ySize && yE > m_yBegin
+                     : yB > m_yBegin && yE < m_yBegin + m_ySize;
+  }
+
+  /// @brief check if the lines cross inside the section
+  /// @brief line1 - functional form of line 1
+  /// @brief line1 - functional form of line 2
+  /// @warning note that this function is assuming that these are lines
+  /// It may be incorrect assumption for rapidly changing function or large
+  /// sections
+  /// @return true if the lines cross in the section
+  template <typename F>
+  inline bool isCrossingInside(F line1, F line2) const {
+    // this microalgorithm idea is illustrated below
+    // section left section right
+    // example with crossing
+    //                                       |            +2
+    // line 1 crossing left section edge     +1          _|
+    // left edge mid point                   |_           |
+    //                                       |            +1
+    // line 2crossing left section           +2           |
+    //
+    // example with no crossing
+    //                                       |            +1
+    // line 1 crossing left section edge     +1          _|
+    // left edge mid point                   |_           |
+    //                                       |            +2
+    // line 2crossing left section           +2           |
+    //
+    // if for any of the two lines the condition
+    // (line1_left_y-middle_on_the_left_y)*(line1_right_y-middle_on_the_right_y)
+    // < 0 means that there is crossing
+
+    float line1_left_y = line1(xBegin());
+    float line1_right_y = line1(xBegin() + xSize());
+    float line2_left_y = line2(xBegin());
+    float line2_right_y = line2(xBegin() + xSize());
+    float left_mid = 0.5f * (line1_left_y + line2_left_y);
+    float right_mid = 0.5f * (line1_right_y + line2_right_y);
+    return (line1_left_y - left_mid) * (line1_right_y - right_mid) < 0;
+  }
+
+  // counter clock wise distance from upper left corner
+  // a and b are line parameters y = ax + b
+  float distCC(float a, float b) const;
+  // anti-counter clock wise distance from upper left corner
+  float distACC(float a, float b) const;
+
+  // sizes
+  float xSize() const { return m_xSize; }
+  float ySize() const { return m_ySize; }
+  float xBegin() const { return m_xBegin; }
+  float yBegin() const { return m_yBegin; }
+  unsigned divisionLevel() const { return m_divisionLevel; }
+
+  /// store additional (arbitrary) info in indexed array
+  /// @param index - identifier
+  /// @param value - value to store
+  void setHistory(unsigned index, float value) {
+    m_history.resize(index + 1);
+    m_history[index] = value;
```

Comment:
```suggestion
    m_history.at(index) = value;
```

If it's not performance critical to skip the bounds check.